procedures/release-duty/index.rst

```diff
@@ -65,25 +65,21 @@ Email
 
 As ReleaseDuty you need to *subscribe* to certain mailing lists:
 
- * All types of sign-offs and approvals should go to `release signoff mailing list <https://mail.mozilla.org/listinfo/release-signoff>`__
- * All discussion topics should go to `release drivers mailing list <https://mail.mozilla.org/listinfo/release-drivers>`__
-
-These other mailing lists will likely have useful discussions and information:
+These mailing lists will likely have useful discussions and information:
  * `RelEng internal mailing list <release@mozilla.com>`__
- * `Thunderbird release drivers mailing list <https://mail.mozilla.org/listinfo/thunderbird-drivers>`__
- * `release-automation-notifications-thunderbird mailing list <https://mail.mozilla.org/listinfo/release-automation-notifications-thunderbird>`__
```

Comment:
This was removed on the 9th of July 2020 as far as I can tell so it's great to clean it up.